pkg/athena/api/api.go

```diff
@@ -57,6 +57,12 @@ func (c *API) Execute(ctx context.Context, input *api.ExecuteQueryInput) (*api.E
 			Database: aws.String(c.settings.Database),
 		},
 		WorkGroup: aws.String(c.settings.WorkGroup),
+		ResultReuseConfiguration: &athena.ResultReuseConfiguration{
+			ResultReuseByAgeConfiguration: &athena.ResultReuseByAgeConfiguration{
+				Enabled:         aws.Bool(c.settings.ResultReuseEnabled),
+				MaxAgeInMinutes: aws.Int64(c.settings.ResultReuseMaxAgeInMinutes),
+			},
+		},
 	}
 
 	if c.settings.OutputLocation != "" {
```

Comment:
Just a note that I think that below here, after we receive the QueryExecutionId we'll be able to get additional information if result reuse was in place or not. See https://github.com/grafana/oss-plugin-partnerships/issues/127

`c.Client.GetQueryExecutionWithContext(ctx, &athena.GetQueryExecutionInput{QueryExecutionId: output.QueryExecutionId})`

Reply:
i had to do the check in the `Status` method. when i did it immediately after getting the query execution id, the boolean that indicated whether it was reusing a query or not was always false for me. probably because it just started the query.